app/controllers/audios_controller.rb

```diff
@@ -20,6 +20,7 @@ def new
     @formdef = PSSBrowserUploads.av_form_definition('audio')
     @accepted_types = %w(.mp3 .mp2 .mp4a .wav .flac .aif .aiff .wma .mpga .oga
                          .ogg .au .adp .aac .weba).join(',')
+    @source = Source.find_by_id(params[:source_id])
```

Comment:
Could be that I'm missing something, but does `Source` need to be queried here, and in the other controllers? It seems that the goal is to pass the ID along to the Javascript that processes the form, so that it can put the source ID into the post data that it submits.

I could see querying Source here if the purpose is to verify that the ID exists; but then that check should be added.